modules/launcher/src/main/java/org/mule/module/launcher/util/ObservableList.java

```diff
@@ -323,14 +324,16 @@ public boolean hasNext()
         @Override
         public E next()
         {
-            cursor++;
-            return iterDelegate.next();
+            E next = iterDelegate.next();
+            lastRet = cursor++;
```

Comment:
The original implementation of set() is	broken. For example, the following simple test code throws IndexOutOfBoundsException. I will add it to ObservableListTestCase.

```
@Test
public void testListIteratorSetAfterNext()
{
    ObservableList<String> list = new ObservableList<String>();                                                                                               
    list.add("a");                                                                                                                                            
    ListIterator<String> it = list.listIterator();                                                                                                            
    String s = it.next();                                                                                                                                     
    it.set("A");                                                                                                                                              

    assertThat("a", equalTo(s));                                                                                                                              
    assertThat("A", equalTo(list.get(0)));                                                                                                                    
}
```

I introduced lastRet to make set() work with both next() and previous().
java.util.Collections#reverse() uses both as below:

```
ListIterator fwd = list.listIterator();                                                                                                                       
ListIterator rev = list.listIterator(size);                                                                                                                   
for (int i=0, mid=list.size()>>1; i<mid; i++) {                                                                                                               
    Object tmp = fwd.next();                                                                                                                                  
    fwd.set(rev.previous());                                                                                                                                  
    rev.set(tmp);                                                                                                                                             
}
```